In [2]:
import numpy as np

# Read the CSV file. The column names become names we can use in NumPy.
data = np.genfromtxt(
    "real_estate_dataset.csv",
    delimiter=",",
    names=True,
    dtype=None,
    encoding="utf-8"
)

print("Number of homes:", len(data))
print("First home:")
print(data[0])

# Get a few columns from the table.
prices = data["Price"]
sizes = data["Square_Feet"]
print("\nFirst five prices:", prices[:5])
print("First five sizes:", sizes[:5])


Number of homes: 500
First home:
(1, 143.63502971184062, 1, 3, 3, 1967, 1, 1, 48, 8.297631202876449, 5.935733640397012, 602134.816746586)

First five prices: [602134.81674659 591425.13538622 464478.69687988 583105.65599648
 619879.14252279]
First five sizes: [143.63502971 287.6785766  232.99848545 199.66462105  89.00466011]


In [3]:
price_per_sqft = prices / sizes

print("PRICE (USD)")
print("Mean:", round(np.mean(prices), 2))
print("Median:", round(np.median(prices), 2))
print("Minimum:", round(np.min(prices), 2))
print("Maximum:", round(np.max(prices), 2))

print("\nSIZE (square feet)")
print("Mean:", round(np.mean(sizes), 2))
print("Median:", round(np.median(sizes), 2))
print("Minimum:", round(np.min(sizes), 2))
print("Maximum:", round(np.max(sizes), 2))

print("\nPRICE PER SQUARE FOOT (USD)")
print("Mean:", round(np.mean(price_per_sqft), 2))
print("Median:", round(np.median(price_per_sqft), 2))
print("Minimum:", round(np.min(price_per_sqft), 2))
print("Maximum:", round(np.max(price_per_sqft), 2))

# The 75th percentile is the price cutoff for the top quarter.
price_cutoff = np.percentile(prices, 75)
print("\n75th percentile price:", round(price_cutoff, 2))


PRICE (USD)
Mean: 582209.63
Median: 574724.11
Minimum: 276892.47
Maximum: 960678.27

SIZE (square feet)
Mean: 174.64
Median: 178.29
Minimum: 51.27
Maximum: 298.24

PRICE PER SQUARE FOOT (USD)
Mean: 4008.82
Median: 3361.19
Minimum: 1636.48
Maximum: 11572.99

75th percentile price: 665942.3


In [4]:
top_mask = prices >= price_cutoff
top_homes = data[top_mask]

print("Homes in the top quarter:", len(top_homes))
print("IDs of the first ten homes in that group:", top_homes["ID"][:10])

# Compare every home's price with the average price.
average_price = np.mean(prices)
above_average = prices > average_price
below_average = prices < average_price

print("\nAverage price:", round(average_price, 2))
print("Homes above average:", np.sum(above_average))
print("Homes below average:", np.sum(below_average))

# Check one example property: the first row in the data.
first_price = prices[0]
if first_price > average_price:
    print("The first home's price is above average.")
else:
    print("The first home's price is below or equal to average.")


Homes in the top quarter: 125
IDs of the first ten homes in that group: [ 6  8  9 12 13 20 21 39 40 46]

Average price: 582209.63
Homes above average: 240
Homes below average: 260
The first home's price is above average.


In [5]:
price_order = np.argsort(prices)
price_order = price_order[::-1]
top_ten = data[price_order[:10]]

print("ID | Size | Bedrooms | Location score | Price")
for home in top_ten:
    print(
        int(home["ID"]), "|",
        round(home["Square_Feet"], 2), "|",
        int(home["Num_Bedrooms"]), "|",
        round(home["Location_Score"], 2), "|",
        round(home["Price"], 2)
    )

print("\nAverage size of these homes:", round(np.mean(top_ten["Square_Feet"]), 2))
print("Average bedrooms of these homes:", round(np.mean(top_ten["Num_Bedrooms"]), 2))
print("Average location score of these homes:", round(np.mean(top_ten["Location_Score"]), 2))


ID | Size | Bedrooms | Location score | Price
151 | 277.07 | 5 | 9.04 | 960678.27
53 | 284.87 | 5 | 7.61 | 909199.56
201 | 210.51 | 5 | 8.18 | 885909.22
8 | 266.54 | 5 | 9.37 | 875352.55
474 | 272.79 | 5 | 6.32 | 867558.3
500 | 296.55 | 4 | 7.89 | 864299.5
108 | 238.89 | 5 | 2.54 | 860013.26
368 | 276.34 | 4 | 3.86 | 857487.18
379 | 253.62 | 5 | 7.43 | 854374.52
70 | 296.72 | 4 | 4.07 | 854304.13

Average size of these homes: 267.39
Average bedrooms of these homes: 4.7
Average location score of these homes: 6.63


## 5. Business insight
The ten highest-priced homes average 267.4 square feet and 4.7 bedrooms, compared with 174.6 square feet and 3.0 bedrooms across all homes. Their average location score is 6.6, compared with 5.2 overall. In this dataset, larger homes and homes with more bedrooms are associated with higher prices; the top ten also have a higher average location score. These numbers describe this dataset and do not prove that one feature causes a higher price